### PR TITLE
Backports 0.16 pr11

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1110,12 +1108,11 @@ bool AppInitParameterInteraction()
         InitWarning(strprintf(_("Reducing -maxconnections from %d to %d, because of system limitations."), nUserMaxConnections, nMaxConnections));
 
     // ********************************************************* Step 3: parameter-to-internal-flags
-
-    if (gArgs.IsArgSet("-debug")) {
+    if (gArgs.IsArgSet("-debug") > 0) {
```
Comment (UdjinM6) [inline src/init.cpp]:
```suggestion
    if (gArgs.IsArgSet("-debug")) {
```

File: src/init.cpp [hunk 2]
```diff
@@ -1805,6 +1805,9 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
                 pcoinsTip.reset();
                 pcoinsdbview.reset();
                 pcoinscatcher.reset();
+                // new CBlockTreeDB tries to delete the existing file, which
+                // fails if it's still open from the previous loop. Close it first:
+                pblocktree.reset();
```
Comment (UdjinM6) [inline src/init.cpp]: Should apply the same fix to `evoDb` and `deterministicMNManager` below I think.

Comment (PastaPastaPasta) [reply src/init.cpp]: https://github.com/dashpay/dash/pull/3361/commits/6c61159a9911ab5e9ed2fe1528b4d92557439d95

**Comment by PastaPastaPasta** [comment]
squashed fix 11309 into 11309

File: src/net_processing.cpp [hunk 3]
```diff
@@ -2516,20 +2517,22 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
...
             }
 
             mempool.check(pcoinsTip.get());
             connman->RelayTransaction(tx);
+
             for (unsigned int i = 0; i < tx.vout.size(); i++) {
                 auto it_by_prev = mapOrphanTransactionsByPrev.find(COutPoint(inv.hash, i));
                 if (it_by_prev != mapOrphanTransactionsByPrev.end()) {
-                    for (const auto& elem : it_by_prev->second) {
+                    for (const auto &elem : it_by_prev->second) {
```
Comment (UdjinM6) [inline src/net_processing.cpp]: Why? This breaks dev styling guide `Align pointers and references to the left i.e. use type& var and not type &var.` https://github.com/dashpay/dash/blob/master/doc/developer-notes.md (not to mention that this and changes to whitespaces are unrelated and do not exist in the original PR).

Comment (PastaPastaPasta) [reply src/net_processing.cpp]: was this present before the squash? (I didn't actively do this, CLion potentially did it without me realizing)

Comment (PastaPastaPasta) [reply src/net_processing.cpp]: applied patch and squashed into 11309
